src/main/java/thecodex6824/thaumicaugmentation/api/augment/builder/impulsecannon/IImpulseCannonConversion.java

```diff
@@ -0,0 +1,123 @@
+/**
+ *  Thaumic Augmentation
+ *  Copyright (c) 2019 TheCodex6824.
+ *
+ *  This file is part of Thaumic Augmentation.
+ *
+ *  Thaumic Augmentation is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Thaumic Augmentation is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with Thaumic Augmentation.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package thecodex6824.thaumicaugmentation.api.augment.builder.impulsecannon;
+
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import org.jetbrains.annotations.NotNull;
```

Comment:
I don't know if we currently have these in the normal build.gradle, although I think just adding it as a dependency should work fine. I also have `@ParametersAreNonnullByDefault` set on the API top-level `package-info.java`, but from the name I guess that doesn't extend to return values.